### PR TITLE
feat: SearchInput component with debounce functionality

### PR DESCRIPTION
**Pull request by charmi-v**
## Description

 **Component**: SearchInput
- Modified the component with new props to manage debounce functionality.
- Added a test file 
- Added required mock files for running jest test
- Added a documentation file describing the component

## Why

**New Feature:** Enhanced the `SearchInput` component with debounce functionality.

## Issue

#222 

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines) 
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by oyo — CHANGES_REQUESTED** [review]
Good contribution - but since we are publishing a new release on every PR merge please provide a short summary at the top of CHANGELOG.md

**Review by oyo — APPROVED** [review]
looks good - thanks for your contribution

File: src/components/basic/SearchInput/SearchInput.test.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react'
```
Comment (oyo) [inline src/components/basic/SearchInput/SearchInput.test.tsx:1]: please add license headers to all newly created files

**Review by oyo — CHANGES_REQUESTED** [review]
@evegufy pointed out missing license headers on all new files

File: src/components/basic/SearchInput/SearchInput.test.tsx (hunk 1 above)
Comment (charmi-v) [reply src/components/basic/SearchInput/SearchInput.test.tsx:1]: okay sure

**Review by oyo — APPROVED** [review]
lgtm now

**Comment by evegufy** [comment]
Hi @charmi-v  could you please resolve the conflicts? I'll merge the pr afterwards. Sorry for the inconvenience.

**Comment by charmi-v** [comment]
> Hi @charmi-v could you please resolve the conflicts? I'll merge the pr afterwards. Sorry for the inconvenience.

Hello @evegufy , 
Conflicts have been resolved, and the package version has been updated to 3.0.32.

**Review by evegufy — APPROVED** [review]

**Pull request merged**